src/Dialogue.js

```diff
@@ -26,42 +35,77 @@ export default class Dialogue {
     if (this.isComplete()) {
       return this.completeText;
     } else if (this.isBattle()) {
-      return `${this.conversation[this.progress].description}\n${this.startBattle()}`;
+      return `${this.activeText.description}\n${this.startBattle()}`;
     }
-    return this.conversation[this.progress].description;
+    return this.activeText.description;
   }
 
   /**
    * Restart the dialogue!
    */
   restart() {
-    this.progress = 0;
+    for (var text in this.conversation) {
+      this.activeText = text;
+      break;
+    }
```

Comment:
Ah yeah I did break that. Progress will have to be changed to the key of the current block.